### ISSUE TITLE
Error downloading content with js [net::ERR_FAILED 200 (OK)]

### ISSUE THREAD

**Issue by ztmaev** [Alex313031/thorium#306]
**System Details**
 - OS: [Ubuntu 22.04]
 - Thorium Version [117.0.5938.157]

**Problem**
This error shows in the console while downloading files using js, opening the file's link on its own tab works fine (downloads the file.)
The page works fine on chrome.

**Screenshots**
![image](https://github.com/Alex313031/thorium/assets/97959215/12c3da42-2ec5-49a7-b14b-20ffd8ea60a4)

**Additional Notes**
_The code producing the error_
```js
function downloadFile(url, fileName) {
        infoText.style.display = 'none';
        infoFooter.innerHTML = 'Downloading, please don\'t close this page.';

        const xhr = new XMLHttpRequest();
        xhr.open("GET", url, true);
        xhr.responseType = "blob";

        xhr.addEventListener("progress", (event) => {
            if (event.lengthComputable) {
                const percentComplete = (event.loaded / event.total) * 100;
                progress_bar.value = percentComplete;
                progress_percent.innerHTML = toFullNumber(percentComplete) + '%';
            }
        });

        xhr.onreadystatechange = () => {
            if (xhr.readyState === 4 && xhr.status === 200) {
                const blob = xhr.response;
                const a = document.createElement("a");
                const url = window.URL.createObjectURL(blob);
                a.href = url;
                a.download = fileName;
                document.body.appendChild(a);
                a.click();
                window.URL.revokeObjectURL(url);
                infoFooter.innerHTML = 'Download completed!';
            }
        };

        xhr.send();
    }

**Comment by gz83** [comment]
Can you provide relevant test links?

**Comment by ztmaev** [comment]
> Can you provide relevant test links?

Its a project i'm working on so the server may be offline sometimes,

https://github.com/Alex313031/thorium/assets/97959215/00a37696-209a-4d55-8b45-c9c044841b59


[https://diskord.maev.site/](url)
You can use this login:
username: demo
pass: Demo!123.

**Comment by gz83** [comment]
![image](https://github.com/Alex313031/thorium/assets/17492865/63930ce1-18b5-420e-8d46-f54be91dc178)

Thank you very much for providing the test cases.

I uploaded a 300M compressed package to your website and conducted upload and download tests. I was unable to reproduce the error you encountered, and I did not receive any error message in devtools.

**Comment by ztmaev** [comment]
> ![image](https://user-images.githubusercontent.com/17492865/277170584-63930ce1-18b5-420e-8d46-f54be91dc178.png)
> 
> Thank you very much for providing the test cases.
> 
> I uploaded a 300M compressed package to your website and conducted upload and download tests. I was unable to reproduce the error you encountered, and I did not receive any error message in devtools.

It seems to be an issue with an extension, Incognito tabs also work fine. I'm sorry

**Comment by KeazoN** [comment]
**What exactly is the solution? Can you explain more clearly?**

_I would like to point out that I have encountered this problem many times and our code (project) structure is the same_

**Comment by nekochanoide** [comment]
@KeazoN or anyone who searching net::ERR_FAILED 200 (OK)
for me was no space left on disk

**Comment by l123wx** [comment]
This might help you a little bit:

https://stackoverflow.com/questions/78587174/chrome-browser-xhr-request-error-neterr-failed-200-on-subsequent-requests

**Comment by AlttiRi** [comment]
The error "net::ERR_FAILED 200 (OK)" also appears when the size of the "blob_storage" folder in your browser profile reaches a certain limit (which depends on the size of free disk space).

I came across this when I downloaded files with DevTools open, inspecting the background script of the extension.

It seems that there is a bug in Chrome, because of which, while DevTools inspects the ~background extension~ script (which executes **XHR**s with `xhr.responseType = "blob";`), the cache in `blob_storage` is not cleared (garbage collected).

In my case, I just needed to close DevTools. 

---

UPD. 
It's happens on common webpages too. 
While DevTools is opened the cache data in "blob_storage" folder from `XMLHttpRequest`s with `xhr.responseType = "blob";` will not be garbage collected.
